### PR TITLE
Systemd for modern RH and apply recipe fix

### PR DESCRIPTION
**Pull request by realloc**
* Use systemd for modern RH and Fedora.
* Always send restart to apply changes immediately

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -54,6 +54,6 @@
     action :nothing
     source 'sysctl.conf.erb'
     mode '0644'
-    notifies :start, 'service[procps]', :immediately
+    notifies :restart, 'service[procps]', :immediately
```
Comment (svanzoest) [inline recipes/default.rb]: I do not think this works for Ubuntu. At least on my Trusty (14.04) system,
`restart procps` fails with `restart: Unknown instance:`. `service procps restart` does seem to work however.

So I am open to this change, but we need to make sure it is backwards compatible.

File: recipes/service.rb [hunk 2]
```diff
@@ -31,8 +31,21 @@
   when 'arch', 'exherbo'
     service_name 'systemd-sysctl'
     provider Chef::Provider::Service::Systemd
+  when 'centos','redhat'
+    if node['platform_version'].to_f >= 7.0
+      service_name 'systemd-sysctl'
+      provider Chef::Provider::Service::Systemd
+    end
+  when 'redora'
```
Comment (asmarre) [inline recipes/service.rb]: Perhaps a misspelling of 'fedora'?

File: recipes/service.rb [hunk 3]
```diff
@@ -31,8 +31,21 @@
   when 'arch', 'exherbo'
     service_name 'systemd-sysctl'
     provider Chef::Provider::Service::Systemd
+  when 'centos','redhat'
```
Comment (ramereth) [inline recipes/service.rb]: Needs a space after the comma.

Also, this fails [FC024](https://acrmp.github.io/foodcritic/#FC024).

**Comment by svanzoest** [comment]
I think this is likely resolved through other fixes, if we missed anything please reopen a new pull request for the missing pieces.

**Pull request closed without merging**